pkg/resource/edit/errors.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package edit
+
+import (
+	"fmt"
+
+	"github.com/pulumi/pulumi/pkg/resource"
+)
+
+// ResourceHasDependenciesError is returned by DeleteResource if a resource can't be deleted due to the presence of
+// resources that depend directly or indirectly upon it.
+type ResourceHasDependenciesError struct {
+	Condemned    *resource.State
+	Dependencies []*resource.State
+}
+
+func (r ResourceHasDependenciesError) Error() string {
+	return fmt.Sprintf("Can't delete resource %q due to dependent resources", r.Condemned.URN)
```

Comment:
urns are hge. Not a fan of them showing up in the midle of things like this.

Reply:
This string is never actually printed anywhere - the CLI presents a much better error message based on the fields of `ResourceHasDependenciesError` instead of using this message.